test/integration/shim_int_test.go

```diff
@@ -0,0 +1,121 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/stretchr/testify/suite"
+)
+
+type ShimIntegrationTestSuite struct {
+	suite.Suite
+}
+
+func (suite *ShimIntegrationTestSuite) createProjectFile(ts *e2e.Session) {
+	ts.PrepareActiveStateYAML(strings.TrimSpace(`
+		project: https://platform.activestate.com/ActiveState-CLI/Python3?commitID=fbc613d6-b0b1-4f84-b26e-4aa5869c4e54
+	`))
+}
+
+func (suite *ShimIntegrationTestSuite) TestShim_Basic() {
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	suite.createProjectFile(ts)
+
+	scriptBlock := `
+print("Hello World!")
+	`
+
+	testScript := filepath.Join(fmt.Sprintf("%s/%s.pl", ts.Dirs.Work, suite.T().Name()))
+	fail := fileutils.WriteFile(testScript, []byte(scriptBlock))
+	suite.Require().NoError(fail.ToError())
+
+	cp := ts.Spawn("shim", "--language", "python3", fmt.Sprintf("%s", testScript))
```

Comment:
I'm not sure what went wrong here. The first AC says:

> `state shim -- perl foo.pl` runs `perl foo.pl` inside our runtime environment

But what you're testing here is

```
state shim --language python3 "Hello World!"
```

Please stick to the AC's.

Reply:
Here's some examples of things that should work with `state shim`, maybe that helps:

```
state shim -- whoami
state shim -- perl foo.pl
state shim -- perlcritic foo.pl
state shim -- find "foo" | grep bar
```

Note the double hyphen is crucial. You don't HAVE to use it, but we HAVE to support it.

Reply:
Thanks, the examples helped.

Reply:
It means we have to support both `state shim whoami` and `state shim -- whoami`. I believe it's simply a matter of setting `cmd.SetDisableFlagParsing(true)`

Reply:
Turns out we have to leave the flag parsing in order for cobra to handle this